### PR TITLE
TCE UI: Adds websocket service and streaming logs for cluster create

### PR DESCRIPTION
**Pull request by miclettej**
## What this PR does / why we need it

Adds a websocket service that wraps @react-use-websocket

Adds app environment awareness (development vs production) to global Store

Adds an example implementation of leveraging websocket service to stream logs of cluster creation

Requires additional PR to include node.js mock server; post separate PR shortly

## Which issue(s) this PR fixes

Fixes: # 3567

## Describe testing done for PR
Verified websocket connection started from UI and streams logs

### CODE REVIEW

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/temp/progress.component.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    console.log(`TODO: output statusMessageHistory as steps complete in UI ${statusMessageHistory}`);
+
+    // Formats a log line to include pre-pended log type (INFO, WARNING, ERROR)
+    const formatLog = (log: any): string => {
+        return `[${log.data.logType}] ${log.data.message}`;
+    };
+
+    // Wraps log line in a span with a custom css class if log type is Error or Warning
+    const setLogLineCssClass = (e: string) => {
```
Comment (raymondz1) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/temp/progress.component.tsx:76]: can we rename the `e`?

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/temp/progress.component.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+// App imports
+import { useWebsocketService, WebsocketService, WsOperations } from '../../shared/services/Websocket.service';
+import './progress.component.scss';
+
+const LogViewContainer = styled.section`
+    height: 500px;
+    width: 900px;
+`;
+
+export const LogTypes = {
```
Comment (raymondz1) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/temp/progress.component.tsx:19]: If we use this logTypes all over the places, I think we should move it to App.constants.tsx or create another constant file for it.

File: cli/cmd/plugin/ui/web/tanzu-ui/src/App.tsx [hunk 3]
```diff
@@ -1,20 +1,38 @@
...
+            // TODO: why can't I seem to apply a type to the generic for 'dispatch' <Action>
+            dispatch({
+                type: APP_ENV_CHANGE,
+                payload: {
+                    name: 'appEnv',
+                    value: process.env.NODE_ENV
+                }
+            });
+        }
+    }, [],);
```
Comment (raymondz1) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/App.tsx:34]: remove second comma `,`

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/temp/progress.component.tsx (hunk 2 above)
Comment (miclettej) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/temp/progress.component.tsx:19]: Yes, I was thinking that we may want to create a component that wraps the LazyLog library, so we can reuse most of what I have done here in other 'deploy progress' pages. For example we will want to show deployment progress for management clusters, workload clusters, and unmanaged clusters. If we create a reusable component, then I think we would define the LogTypes in that reusable component. I do not think we will use LogTypes in other parts of the app. Does that sound good?

Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/temp/progress.component.tsx:20]: Yes. It makes sense. Create a new wrapper component for the LayLog is a good idea.

**Review by raymondz1 — APPROVED** [review]
LGTM

**Pull request merged**